docs/contributing/featureflags.md

```diff
@@ -19,35 +19,44 @@ highlights:
 - Environments (production, QA, and development for now) exist to segment flag states further. This
   will be automatic based on where code is running.
 
-## Using feature flags in code
+## Flag data sources
 
-Default to an "off" state whenever possible – code defensively so that existing functionality is
-maintained should a flag be unavailable altogether. When an interface supports it, also provide
-default values implying "off" to feature flag accessors.
+When consuming feature flags in either the client or server code, it is important to understand
+where the flags are sourced.
 
-Offline mode makes default values even more important, and local development as well as self-hosted
-installations imply being offline. Set a safe default value not just in the flag definition online
-at LaunchDarkly but in code.
+The source of the flags is dependent upon the Bitwarden server instance that is being used, as for
+client development the flags are served from the Bitwarden `Api`.
 
-### Server
+| Server configuration | Flag source                                            |
+| -------------------- | ------------------------------------------------------ |
+| Local development    | Local `flags.json` file                                |
+| Self-hosted          | Flags are "off", unless local `flags.json` is provided |
+| QA Cloud             | LaunchDarkly QA                                        |
+| Production Cloud     | LaunchDarkly Production                                |
 
-1. Inject `IFeatureService` where you need a feature flag. Note that you’ll also need
-   `ICurrentContext` when accessing the feature state.
-2. Check the
-   [FeatureFlagKeys](https://github.com/bitwarden/server/blob/master/src/Core/Constants.cs) constant
-   list for the key you plan on using, and add it if not found.
-3. Utilize the above key constant with the appropriate method on the feature service:
+:::note Self-hosted support
```

Comment:
💭 on what level this should be, note or warning. I think this is fine though.

Reply:
I was back and forth on that as well, but changing it to Warning does seem appropriate.  I've changed it.